### PR TITLE
Add DidWeb::create() and DidWeb::resolve(), add unit tests, bind to kt

### PR DESCRIPTION
**Pull request by KendallWeihe**

### CODE REVIEW

File: bound/kt/src/test/kotlin/web5/sdk/dids/methods/web/DidWebTests.kt [hunk 1]
```diff
@@ -1,50 +1,276 @@
...
+
+            // temporarily removing @context from document, need to rework Document type
+            bearerDid.document.context = null
+
+            mockWebServer.enqueue(
+                MockResponse()
+                    .setResponseCode(200)
+                    .addHeader("Content-Type", "application/json")
+                    .setBody(Json.stringify(bearerDid.document))
+            )
```
Comment (diehuxx) [inline bound/kt/src/test/kotlin/web5/sdk/dids/methods/web/DidWebTests.kt:265]: We could stricten this test by only returning did doc from `/.well-known/did.json `. If you wanna go bananas, we could add a test case for when did uri specifies the path

File: docs/API_DESIGN.md [hunk 2]
```diff
@@ -624,27 +623,24 @@ CLASS DidJwkCreateOptions
...
 >
-> The `CONSTRUCTOR(domain: string, public_jwk: Jwk)` does not publish the DID Document to a host, but merely creates the instance of the `did:web` in the local scope. 
+> The `create` does not publish the DID Document to a host, but creates the instance of the `did:web` `BearerDid` in the local scope. 
 
 ```pseudocode!
 CLASS DidWeb
-  CONSTRUCTOR(domain: string, public_jwk: Jwk)
-  CONSTRUCTOR(uri: string)
+  STATIC METHOD create(domain: string, options: DidWebCreateOptions?): BearerDid
   STATIC METHOD resolve(uri: string): ResolutionResult
```
Comment (diehuxx) [inline docs/API_DESIGN.md:631]: outside the scope of this PR, but curious if you've given thought to whether did web resolve, did dht resolve, and did dht publish should be async?

**Review by diehuxx — APPROVED** [review]

**Review by nitro-neal — APPROVED** [review]

File: docs/API_DESIGN.md (hunk 2 above)
Comment (KendallWeihe) [reply docs/API_DESIGN.md:631]: yeah I think that's right actually, we should do that ... async programming in the binded world maybe needs a bit more thought 🤔 

https://github.com/TBD54566975/web5-rs/issues/309

File: bound/kt/src/test/kotlin/web5/sdk/dids/methods/web/DidWebTests.kt (hunk 1 above)
Comment (KendallWeihe) [reply bound/kt/src/test/kotlin/web5/sdk/dids/methods/web/DidWebTests.kt:265]: yeah, I agree with both of those items! but for a later time 😅

**Pull request merged**